lib/grammers-client/src/types/inline_query.rs

```diff
@@ -0,0 +1,169 @@
+use crate::{client::Client, utils::generate_random_id, InputMessage};
+use grammers_mtsender::InvocationError;
+use grammers_tl_types as tl;
+
+/// Represents an inline query update, which occurs when you sign in as a bot and a user sends an
+/// inline query such as `@bot query`.
+pub struct InlineQuery {
+    query: tl::types::UpdateBotInlineQuery,
+    client: Client,
+}
+
+/// An inline query answer builder.
+pub struct Answer {
+    request: tl::functions::messages::SetInlineBotResults,
+    client: Client,
+}
+
+impl InlineQuery {
+    pub(crate) fn new(client: &Client, query: tl::types::UpdateBotInlineQuery) -> Self {
+        Self {
+            query,
+            client: client.clone(),
+        }
+    }
+
+    // User that sent the query.
+    pub fn user_id(&self) -> i32 {
+        self.query.user_id
+    }
+
+    // The text of the inline query.
+    pub fn text(&self) -> &str {
+        self.query.query.as_str()
+    }
+
+    /// Answer the inline query.
+    // TODO: add example
+    pub fn answer(&self, results: Vec<tl::enums::InputBotInlineResult>) -> Answer {
+        Answer {
+            request: tl::functions::messages::SetInlineBotResults {
+                gallery: false,
+                private: false,
+                query_id: self.query.query_id,
+                results,
+                cache_time: 0,
+                next_offset: None,
+                switch_pm: None,
+            },
+            client: self.client.clone(),
+        }
+    }
+}
+
+impl Answer {
+    /// Whether the results should show as a gallery (grid) or not.
+    /// Defaults to false.
+    pub fn gallery(mut self, gallery: bool) -> Self {
```

Comment:
Do we want a `bool` input parameter on this one, or would it be enough to have it parameter-less and calling it sets it to `true`?

I guess it depends. How often will `gallery` be determined programmatically? If it's not a lot, `.gallery()` may be more comfortable to write than `.gallery(true)`. (The same applies to `private`.)

Reply:
I've been thinking about that too, perhaps calling `.gallery()` is the better way to go. (I really wish Rust supported optional arguments, it would make things a lot easier!)